modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc

```diff
@@ -469,10 +473,13 @@ TEST(TestSuite, CheckDoInference) {
   test_obj.pclToArray(pcl_pc_ptr, points_array, NORMALIZING_FACTOR);
 
   std::vector<float>* out_detections(new std::vector<float>());
-  test_obj.DoInference(points_array, pcl_pc_ptr->size(), out_detections);
+  std::vector<int>* out_labels(new std::vector<int>());
```

Comment:
Use smart pointers such as uniq_ptr or shared_ptr for safe memory management. Please fix it for all pointer usages.

Reply:
OK. I will refactor part of the code cause it cannot meet our requirements for customizing some of the params. The above problems will be resolved during the refactoring.